[Optix][CUDA] NVRTC compiler backend fails to locate `cuda_fp16.h` and just returns SLANG_FAIL

Slang only looks for `cuda_fp16.h` in `/usr/include` (see `_findCUDAIncludePath`), but Ubuntu puts CUDA headers into `/usr/local/cuda-x[.x]`. The search should probably be expanded to more generically be able to search for potential locations (there might be different ones for other OSes) and then pick the most recent one.

More importantly, Slang shouldn't just return a `SLANG_FAIL` error but also return some diagnostics to inform users of the failure. Now, there is no way around stepping through code in the debugger to figure out why the backend compiler fails.

@mkeshavaNV , @skallweitNV  - Not sure what is remaining here. 

Can you share the repro setup ? 
I tried with slangpy and it does spit out issue if header is not there
```
E       RuntimeError: Failed to create device!
E       C:\Users\local-haaggarwal\slang\slangpy\src\sgl\device\device.cpp:219 in function Device

slangpy\testing\helpers.py:210: RuntimeError
-------------------------------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------------------------------
[INFO] (rhi) driver: [KNOBS]: All knobs on default.

-------------------------------------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------------------------------------
[ERROR] (rhi) layer: CUDA headers not found! Please ensure that the CUDA Toolkit is installed.
Searched for headers (cuda_runtime.h) in following directories:
- C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v13.0\bin\include

====================================================================================================== short test summary info =======================================================================================================
FAILED slangpy/tests/device/test_shader_cursor.py::test_shader_cursor[False-DeviceType.cuda] - RuntimeError: Failed to create device!
FAILED slangpy/tests/device/test_shader_cursor.py::test_shader_cursor[True-DeviceType.cuda] - RuntimeError: Failed to create device!
```

--- a/source/compiler-core/slang-nvrtc-compiler.cpp
+++ b/source/compiler-core/slang-nvrtc-compiler.cpp
@@ -1092,7 +1092,7 @@ SlangResult NVRTCDownstreamCompiler::compile(
     {
         diagnostics->setResult(SLANG_FAIL);
         *outArtifact = artifact.detach();
-        return SLANG_OK;
+        return SLANG_FAIL;
     }
 
     // Neither of these options are strictly required, for general use of nvrtc,
@@ -1175,7 +1175,7 @@ SlangResult NVRTCDownstreamCompiler::compile(
         {
             diagnostics->setResult(SLANG_FAIL);
             *outArtifact = artifact.detach();
-            return SLANG_OK;
+            return SLANG_FAIL;
         }
     }
 

--- a/source/slang/slang-code-gen.cpp
+++ b/source/slang/slang-code-gen.cpp
@@ -1012,12 +1012,16 @@ SlangResult CodeGenContext::emitWithDownstreamForEntryPoints(ComPtr<IArtifact>& 
     // Compile
     ComPtr<IArtifact> artifact;
     auto downstreamStartTime = std::chrono::high_resolution_clock::now();
-    SLANG_RETURN_ON_FAIL(compiler->compile(options, artifact.writeRef()));
+    SlangResult compileResult = compiler->compile(options, artifact.writeRef());
     auto downstreamElapsedTime =
         (std::chrono::high_resolution_clock::now() - downstreamStartTime).count() * 0.000000001;
     getSession()->addDownstreamCompileTime(downstreamElapsedTime);
 
+    // Extract diagnostics regardless of compile result
     SLANG_RETURN_ON_FAIL(passthroughDownstreamDiagnostics(getSink(), compiler, artifact));
+
+    // Now check if compile failed
+    SLANG_RETURN_ON_FAIL(compileResult);
 
     // Copy over all of the information associated with the source into the output
     if (sourceArtifact)

